shelley/chain-and-ledger/executable-spec/src/Cardano/Ledger/Val.hs

```diff
@@ -13,43 +13,38 @@
 
 -- | This module defines a generalised notion of a "value" - that is, something
 -- with which we may quantify a transaction output.
---
--- This module is intended for qualified import:
--- > import qualified Cardano.Ledger.Val as Val
 module Cardano.Ledger.Val
   ( Val (..),
-    (<+>),
-    (<->),
-    (<×>),
+    scale,
     invert,
     sumVal,
     scaledMinDeposit,
-
-    -- * Re-exports
-    (Data.PartialOrd.<=),
-    (Data.PartialOrd.>=),
-    (Data.PartialOrd.==),
-    (Data.PartialOrd./=),
-    (Data.PartialOrd.>),
-    (Data.PartialOrd.<),
-    Data.PartialOrd.compare,
   )
 where
 
-import Data.Group (Abelian, invert)
-import Data.PartialOrd hiding ((==))
-import qualified Data.PartialOrd
+import Data.Group (Abelian)
 import Shelley.Spec.Ledger.Coin (Coin (..))
 
 class
   ( Abelian t,
-    Eq t,
-    PartialOrd t
+    Eq t
   ) =>
   Val t
   where
-  -- | Multiply the value by a scalar
-  scale :: Integral i => i -> t -> t
+  -- | the value with nothing in it
+  zero :: t
+  zero = mempty
```

Comment:
I would say it's better to leave these outside of the class, since they should not vary independently of the monoid values (what would that even mean?)

Reply:
I want them to have value specific types.  Much better error messages. One can always use the Albelian operators if one want to confuse readers of the program.

Reply:
You can still have value specific types, but declaring them outside the class (as they were before) guarantees that they will always agree with the `Monoid` instance.